packages/patrol/android/src/test/kotlin/pl/leancode/patrol/ContractsExtensionsTest.kt

```diff
@@ -0,0 +1,102 @@
+    fun `listDartFilesFlat() lists the Dart test files in the integration_test directory`() {
+
+        // ── permissions
+        // ├── permissions_location_test.dart
+        // └── permissions_many_test.dart
+        // ── sign_in
+        // ├── sign_in_email_test.dart
+        // ├── sign_in_facebook_test.dart
+        // └── sign_in_google_test.dart
+        // ── example_test.dart
+        val dartTestGroup = dartTestGroup {
+            name = "root"
+            groups += listOf(
+                dartTestGroup {
+                    name = "permissions"
+                    groups += listOf(
+                        dartTestGroup {
+                            name = "permissions_location_test.dart"
+                            tests += listOf(
+                                dartTestCase {
+                                    name = "accepts location permission"
+                                }
+                            )
+                        },
+                        dartTestGroup {
+                            name = "permissions_many_test.dart"
+                            tests += listOf(
+                                dartTestCase {
+                                    name = "accepts various permissions"
+                                }
+                            )
+                        }
+                    )
+                },
+                dartTestGroup {
+                    name = "sign_in"
+                    groups += listOf(
+                        dartTestGroup {
+                            name = "sign_in_email_test.dart"
+                            tests += listOf(
+                                dartTestCase {
+                                    name = "signs in with email"
+                                }
+                            )
+                        },
+                        dartTestGroup {
+                            name = "sign_in_facebook_test.dart"
+                            tests += listOf(
+                                dartTestCase {
+                                    name = "signs in with facebook"
+                                }
+                            )
+                        },
+                        dartTestGroup {
+                            name = "sign_in_google_test.dart"
+                            tests += listOf(
+                                dartTestCase {
+                                    name = "signs in with google"
+                                }
+                            )
+                        }
+                    )
+                },
+                dartTestGroup {
+                    name = "example_test.dart"
+                    tests += listOf(
+                        dartTestCase {
+                            name = "some example test"
+                        }
+                    )
+                }
+            )
+        }
+
+        val dartTestFiles = dartTestGroup.listFlatDartFiles()
```

Comment:
In this PR – it did not. I postponed it for later.

[Here](https://github.com/leancodepl/patrol/blob/develop/packages/patrol/android/src/test/kotlin/pl/leancode/patrol/ContractsExtensionsTest.kt)'s what this test looks like now (and [here's the commit that changed it](https://github.com/leancodepl/patrol/commit/af885fa8f3391d5269836dae5b1919affac3ad86))

BTW, It might not be the most fortunate use of the term "flattening" (or maybe even _wrong use_).

By "flattening" here I mean that even if the structure of Dart test files under `integration_test` was tree-like, e.g.:

```
integration_test
└── features
    ├── add_car
    │   ├── in_stock
    │   │   ├── add_car_in_stock_test.dart
    │   │   ├── add_car_in_stock_with_documents_test.dart
    │   │   └── add_car_in_stock_with_financing_test.dart
    │   └── new
    │       └── add_new_car_test.dart
```

then `listFlatDartFiles()` returns the following:

```
integration_test.features.auth.logout.logout_test
integration_test.features.auth.register.register_test
integration_test.features.auth.login.login_test
integration_test.features.add_car.in_stock.add_car_in_stock_test
integration_test.features.add_car.in_stock.add_car_in_stock_with_documents_test
integration_test.features.add_car.in_stock.add_car_in_stock_with_financing_test
integration_test.features.add_car.new.add_new_car_test
integration_test.features.onboarding.onboarding_france_test
```

(That's because if `slash` occurs in a test name in JUnit, dragons appear)



(btw, the above was obtained with `$ find * -type f | grep _test.dart  | xargs -I {} echo integration_test/{} | tr '/' '.' | rev | cut -c 6- | rev`)

I'm very open to a better name.